python/services/dataservice/dmod/dataservice/initial_data_adder_impl.py

```diff
@@ -52,6 +55,8 @@ def add_initial_data(self):
         DmodRuntimeError
             Raised when initial data could not be assembled and/or added successfully to the dataset.
         """
+        original_domain = self._dataset_manager.datasets[self._dataset_name].data_domain
```

Comment:
I am on board with talking about changes to this (though separately).   I'm pretty sure we'd need both `get_dataset` and `has_dataset` functions (at least), which in some ways seems like it might be _more_ details that need to be known, but it would also give us better encapsulation and information hiding.

Aside from that, I already have some other things on my TODO list for changing how initial data adders and datasets interact, as I agree that the coupling isn't very well set up.